### ISSUE TITLE
Dependency for Microsoft.Data.SqlClient (sni) missing

### ISSUE THREAD

**Issue by genifycom** [dotnet/SqlClient#595]
### Describe the bug
Creating a DotNet 3.1 application using Microsoft.Data.SqlClient
Resolver unable to find 'runtime.win-x64.runtime.native.System.Data.SqlClient.sni', version: '4.4.0'
Tried backing Microsoft.Data.SqlClient from 1.1.3 to 1.1.2 and 1.1.1 
Tried installing Nuget runtime.native.System.Data.SqlClient.sni version 4.4.0 and 4.7.0 but same issue.

AWS .NET Core 3.1 Mock Lambda Test Tool (0.10.0)
Unknown error occurred causing process exit: Dependency resolution failed for component xxx\bin\Debug\netcoreapp3.1\xxx.dll with error code -2147450740. Detailed error: Error:
  An assembly specified in the application dependencies manifest (xxx.deps.json) was not found:
    package: 'runtime.win-x64.runtime.native.System.Data.SqlClient.sni', version: '4.4.0'
    path: 'runtimes/win-x64/native/sni.dll'

   at System.Runtime.Loader.AssemblyDependencyResolver..ctor(String componentAssemblyPath)
   at Amazon.Lambda.TestTool.Runtime.LambdaAssemblyLoadContext..ctor(String lambdaPath) in C:\codebuild\tmp\output\src142363207\src\Tools\LambdaTestTool\src\Amazon.Lambda.TestTool\Runtime\LambdaAssemblyLoadContext.cs:line 28
   at Amazon.Lambda.TestTool.Runtime.LocalLambdaRuntime.Initialize(String directory, IAWSService awsService) in C:\codebuild\tmp\output\src142363207\src\Tools\LambdaTestTool\src\Amazon.Lambda.TestTool\Runtime\LocalLambdaRuntime.cs:line 71
   at Amazon.Lambda.TestTool.Runtime.LocalLambdaRuntime.Initialize(String directory) in C:\codebuild\tmp\output\src142363207\src\Tools\LambdaTestTool\src\Amazon.Lambda.TestTool\Runtime\LocalLambdaRuntime.cs:line 46
   at Amazon.Lambda.TestTool.TestToolStartup.Startup(String productName, Action`2 uiStartup, String[] args, RunConfiguration runConfiguration) in C:\codebuild\tmp\output\src142363207\src\Tools\LambdaTestTool\src\Amazon.Lambda.TestTool\TestToolStartup.cs:line 77

### To reproduce
Add AWSTookit to Visual Studio 2019.
Create a Lambda function.
Add Microsoft.Data.SqlClient.
Try to launch the project.

### Expected behavior
Microsoft.Sql.Client dependencies should be all resolved/

### Further technical details
Microsoft.Data.SqlClient version: v1.1.3
.NET target: 3.1
SQL Server version: SQL Server 2016
Operating system: Windows

**Additional context**

**Comment by cheenamalhotra** [comment]
@genifycom 

Are you getting "Access is denied" somewhere internally?
Similar issue: #496

**Comment by cheenamalhotra** [comment]
FYI, I tried installing AWS Toolkit and created Lambda function, did not face any issues compiling project. We don't develop and test external tools generally but dependencies seem to be resolved.

**Comment by genifycom** [comment]
Hi Cheena,
I do not see any access denied.
Were you able to launch the project you created?
I do not have any issues with compiling.
Thanks

**Comment by cheenamalhotra** [comment]
I don't have configurations for AWS credentials, so I can't launch it.

**Comment by tai-yi** [comment]
**Error shows on execute single file after upgrate to `Microsoft.Data.SqlClient 2.0.0` from `1.1.1`**



single file publish command
```
dotnet publish  -r win-x64 -p:PublishSingleFile=true -p:PublishTrimmed=false -o ../bin/win-64 -c release
```

```
Error:
  An assembly specified in the application dependencies manifest (xxx.deps.json) was not found:
    package: 'Microsoft.Data.SqlClient.SNI.runtime', version: '2.0.0'
    path: 'runtimes/win-x64/native/Microsoft.Data.SqlClient.SNI.pdb'
```

**Comment by cheenamalhotra** [comment]
Hi @tai-yi 

Please view issue in dotnet-sdk: https://github.com/dotnet/sdk/issues/10523 and possible solution as mentioned here: https://github.com/dotnet/SqlClient/issues/385#issuecomment-647713950

**Comment by tai-yi** [comment]
@cheenamalhotra  thank you.

**Comment by genifycom** [comment]
Hi Cheena,

I didn't see this one about Microsoft.Data.SqlClient 2.0.0 and I opened a new one. Sorry. It is here https://github.com/dotnet/core/issues/4877

I was already using IncludeSysmbolsInSingleFile and I tried both true and false but I cannot use the Microsoft.Data.SqlClient 2.0.0 in publish single file mode. I have had to use publishsinglefile false.

The error is:
An assembly specified in the application dependencies manifest (MyApp.deps.json) was not found:
package: 'Microsoft.Data.SqlClient.SNI.runtime', version: '2.0.0'
path: 'runtimes/win-x64/native/Microsoft.Data.SqlClient.SNI.pdb'

So something has changed.
Thanks
